app/controllers/ips_controller.rb

```diff
@@ -1,11 +1,13 @@
 class IpsController < ApplicationController
   def new
     @ip = Ip.new
+    @locations = current_locations
   end
 
   def create
-    default_location = current_organisation.locations.first
-    @ip = default_location.ips.new(ip_params)
+    @ip = Ip.new(ip_params)
+    @ip.location = get_location || create_location
```

Comment:
Can this all be rewritten as: 
```
  @ip = Ip.new(params)
  if @ip.save
    publish_for_performance_platform
    redirect_to somewhere
  else
    render new
end


Where params are either:
`{ address: "1.1.1.1", location_attributes: { address: "Made Towers", postcode: "SW12 7YC", organisation_id: 1 } }`
or
`{ address: "1.1.1.1", location_id: 1 }`

Reply:
Also need to add the following to `ip.rb` model:

```
  accepts_nested_attributes_for :location
```

Reply:
What will happen if params are not either but rather combined?

Because without JS, there's no way to stop someone from selecting a location but then still putting stuff in textbox.

`{ address: "1.1.1.1", location_id: 1, location_attributes: { address: "Made Towers", postcode: "SW12 7YC", organisation_id: 1 } }`

Reply:
```irb(main):008:0> Location.last
  Location Load (1.2ms)  SELECT  `locations`.* FROM `locations` ORDER BY `locations`.`id` DESC LIMIT 1
=> #<Location id: 3, radius_secret_key: "C043947A5C5C504A8486", address: "Made Towers", postcode: "SW12 7YC", organisation_id: 1, created_at: "2018-09-20 16:04:30", updated_at: "2018-09-20 16:04:30">
irb(main):009:0> @ip = Ip.new({ address: "1.1.1.1", location_id: 3, location_attributes: { address: "Made Towers", postcode: "SW12 7YC", organisation_id: 1 } })
  Location Load (1.3ms)  SELECT  `locations`.* FROM `locations` WHERE `locations`.`id` = 3 LIMIT 1
=> #<Ip id: nil, address: "1.1.1.1", created_at: nil, updated_at: nil, location_id: nil>
irb(main):010:0> @ip.save
   (0.8ms)  BEGIN
  Organisation Load (1.1ms)  SELECT  `organisations`.* FROM `organisations` WHERE `organisations`.`id` = 1 LIMIT 1
  Location Create (1.0ms)  INSERT INTO `locations` (`radius_secret_key`, `address`, `postcode`, `organisation_id`, `created_at`, `updated_at`) VALUES ('2EF7E0831A2CA97D195D', 'Made Towers', 'SW12 7YC', 1, '2018-09-20 16:05:16', '2018-09-20 16:05:16')
  Ip Create (0.8ms)  INSERT INTO `ips` (`address`, `created_at`, `updated_at`, `location_id`) VALUES ('1.1.1.1', '2018-09-20 16:05:16', '2018-09-20 16:05:16', 4)
   (1.6ms)  COMMIT
=> true
irb(main):011:0> @ip.location
=> #<Location id: 4, radius_secret_key: "2EF7E0831A2CA97D195D", address: "Made Towers", postcode: "SW12 7YC", organisation_id: 1, created_at: "2018-09-20 16:05:16", updated_at: "2018-09-20 16:05:16">```